### ISSUE TITLE
Pure negative queries like -field:xxx being inverted

### ISSUE THREAD

**Issue by gaffinet** [sepinf-inc/IPED#182]
Hi

I found an example with a bug in the free text query interface

when i query **-type:slack** I only get the slack files instead of excluding them.

![grafik](https://user-images.githubusercontent.com/9380336/86485611-73c12c00-bd59-11ea-8a25-ce6908519b92.png)

but, when I query for **\*:\* AND -type:slack** I'm able to properly exclude slack files 

![grafik](https://user-images.githubusercontent.com/9380336/86485868-25f8f380-bd5a-11ea-94c8-8a3e981efe88.png)

**Comment by lfcnassif** [comment]
Actually I don't think this is a bug, but lack of support. Lucene doesn't support pure negative queries, see:

https://cwiki.apache.org/confluence/display/solr/NegativeQueryProblems
https://stackoverflow.com/questions/634765/using-or-and-not-in-solr-query

We can add an example with *:* in the quick search help and maybe try to workaround top level negative queries like Sorl does.

**Comment by gaffinet** [comment]
When I look at the log it seams that the - (minus) is removed somehow

```
2020-07-03 19:54:09	[INFO]	[indexer.desktop.PesquisarIndice]			Searching for query type:slack
2020-07-03 19:54:14	[INFO]	[indexer.desktop.PesquisarIndice]			Searching for query type:slack
2020-07-03 19:54:20	[INFO]	[indexer.desktop.PesquisarIndice]			Searching for query +*:* -type:slack
```
I performed following queries in the interface:

1. type:slack
2. -type:slack
3. *:* AND -type:slack

**Comment by lfcnassif** [comment]
hum... actually from the docs I understood lucene would return nothing instead of negating the negative query. Will investigate...

**Comment by lfcnassif** [comment]
Found no issues in iped code. After upgrading to lucene-5.5.5 to test, pure negative queries started to return no results (expected behaviour), so this is a bug in old lucene-4.x (tried last 4.10.4). We plan to upgrade to Lucene-8.x in #179

**Comment by lfcnassif** [comment]
Just pushed lucene_upgrade branch with initial upgrade to 5.5.5, back compatible with lucene-4.x, so we can include in 3.18.1 after more testing.

**Comment by lfcnassif** [comment]
Fixed in 17babcd2a9099ed21e333e9012da7d19873c5fce also added basic support for pure negative queries, see #188

**Comment by lfcnassif** [comment]
Thank you @gaffinet for reporting!
